packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -151,6 +151,7 @@ function coerceRef(
       if (
         current !== null &&
         current.ref !== null &&
+        typeof current.ref === 'function' &&
```

Comment:
Only attempt to read `_stringRef` from function refs. Otherwise we already know it won't match up.